youtube_dl/downloader/common.py

```diff
@@ -344,8 +346,22 @@ def download(self, filename, info_dict):
 
         sleep_interval = self.params.get('sleep_interval')
         if sleep_interval:
-            self.to_screen('[download] Sleeping %s seconds...' % sleep_interval)
-            time.sleep(sleep_interval)
+            min_sleep_interval = sleep_interval
+            max_sleep_interval = self.params.get('max_sleep_interval')
+
+            try:
+                if max_sleep_interval:
+                    sleep_time = random.uniform(
+                        float(min_sleep_interval),
+                        float(max_sleep_interval)
+                    )
+                else:
+                    sleep_time = float(min_sleep_interval)
+
+                self.to_screen('[download] Sleeping %s seconds...' % sleep_time)
+                time.sleep(sleep_time)
+            except (ValueError, IOError) as err:
+                raise InvalidSleepTimeError()
```

Comment:
I've already clarified there **must be no runtime checks here.** Sleep intervals are static information thus **must be validated on start**.